ts/Data/Connectors/JSONConnector.ts

```diff
@@ -139,14 +137,15 @@ class JSONConnector extends DataConnector {
                     ) :
                     data || []
             )
-            .then((data): Promise<Array<Array<number|string>>> => {
+            .then(async (data): Promise<Array<Array<number|string>>> => {
                 if (data) {
+                    // If already loaded, clear the current rows
+                    table.deleteColumns();
                     converter.parse({ data });
                     table.setColumns(converter.getTable().getColumns());
                 }
-                return connector
-                    .setModifierOptions(dataModifier)
-                    .then((): Array<Array<number|string>> => data);
+                await connector.setModifierOptions(dataModifier);
+                return data;
```

Comment:
No async/await in the data layer as it does not work with the polyfill for old browsers.

Reply:
@bre1470 Is it the es5 builds that don't work? It would be nice to be able to do async/await.

Reply:
Exactly, ES5 will have problems as this is a language feature and not API.